### ISSUE TITLE
CVE-2022-23648 in trivy (gobinary)

### ISSUE THREAD

**Issue by yogeshlonkar** [aquasecurity/trivy#2350]
## Description

trivy gobinary has a CVE-2022-23648 vulnerability

## What did you expect to happen?

Scan to exit without any errors

## What happened instead?

`trivy image xxxx` exits with error

## Output of run with `-debug`:

```
2022-06-17T15:02:52.918+0200    DEBUG   Severities: UNKNOWN,LOW,MEDIUM,HIGH,CRITICAL
2022-06-17T15:02:52.958+0200    DEBUG   cache dir:  ~/Caches/trivy
2022-06-17T15:02:52.959+0200    DEBUG   Skipping DB update...
2022-06-17T15:02:52.959+0200    DEBUG   DB Schema: 2, UpdatedAt: 2022-06-17 12:07:57.264872778 +0000 UTC, NextUpdate: 2022-06-17 18:07:57.264872578 +0000 UTC, DownloadedAt: 2022-06-17 12:42:51.724811 +0000 UTC
2022-06-17T15:02:52.960+0200    INFO    Vulnerability scanning is enabled
2022-06-17T15:02:52.960+0200    DEBUG   Vulnerability type:  [os library]
2022-06-17T15:02:52.992+0200    DEBUG   Image ID: sha256:73cbd7af58c7f1260828b213ce003a7d90fb88500099eeb04e91a17fb3c5ba2f
2022-06-17T15:02:52.992+0200    DEBUG   Diff IDs: [sha256:4fc242d58285699eca05db3cc7c7122a2b8e014d9481f323bd9277baacfa0628 sha256:053183c14d294c76df5746392c0bc5913a86b53871b274051c956f4df2aeace6 sha256:e25cca6435da415c34178cffd526a2e17c6a207ca0b692c1c6bc544dc141ecee sha256:0a294a3cdce0aed1587fec63af41fe16db7015fb870c4ea2bf6ca4ec3eec3527 sha256:89ecb14e7700da0250336ec2d1bfcbc76cda5959b60d5055938422a3bc1f3dca sha256:b2d5f897034569b4ccd80177446796fa4d82d8cbe4be87634ddea65689cba2bf sha256:ae49dcafe2d9cbb6a8b76b9fb516fc80a95e2b03acd7e50c86d9fab618b9a135 sha256:0ee0c99d1f269ca65136cf3808d7ecb418e0f46456111dc7bae4eb1331e4284a sha256:2727042af8b3c3dd6142238fe36853387ab528253c72beedf87a5a689bd111de sha256:cf74e742da146b9c90c4a14e7783411eeee65383a6b53b1376f63998df75620f]
2022-06-17T15:02:52.992+0200    DEBUG   Base Layers: [sha256:4fc242d58285699eca05db3cc7c7122a2b8e014d9481f323bd9277baacfa0628]
2022-06-17T15:02:52.996+0200    INFO    Detected OS: alpine
2022-06-17T15:02:52.996+0200    INFO    Detecting Alpine vulnerabilities...
2022-06-17T15:02:52.996+0200    DEBUG   alpine: os version: 3.15
2022-06-17T15:02:52.996+0200    DEBUG   alpine: package repository: 3.15
2022-06-17T15:02:52.996+0200    DEBUG   alpine: the number of packages: 73
2022-06-17T15:02:53.000+0200    INFO    Number of language-specific files: 2
2022-06-17T15:02:53.000+0200    INFO    Detecting gobinary vulnerabilities...
2022-06-17T15:02:53.000+0200    DEBUG   Detecting library vulnerabilities, type: gobinary, path: trivy
2022-06-17T15:02:53.002+0200    DEBUG   Detecting library vulnerabilities, type: gobinary, path: xxx
2022-06-17T15:02:53.003+0200    DEBUG   Found an ignore file .trivyignore
2022-06-17T15:02:53.003+0200    DEBUG   These IDs will be ignored: []
2022-06-17T15:02:53.003+0200    DEBUG   Found an ignore file .trivyignore
2022-06-17T15:02:53.003+0200    DEBUG   These IDs will be ignored: []
2022-06-17T15:02:53.003+0200    DEBUG   Found an ignore file .trivyignore
2022-06-17T15:02:53.003+0200    DEBUG   These IDs will be ignored: []

xxx (alpine 3.15.4)
================================
Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)


xxx (gobinary)
=====================================
Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)


trivy (gobinary)
================
Total: 1 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 1, CRITICAL: 0)

┌──────────────────────────────────┬────────────────┬──────────┬──────────────────────────────────────┬───────────────────────┬────────────────────────────────────────────────┐
│             Library              │ Vulnerability  │ Severity │          Installed Version           │     Fixed Version     │                     Title                      │
├──────────────────────────────────┼────────────────┼──────────┼──────────────────────────────────────┼───────────────────────┼────────────────────────────────────────────────┤
│ github.com/containerd/containerd │ CVE-2022-23648 │ HIGH     │ v1.6.1-0.20220606171923-c1bcabb45419 │ 1.4.13, 1.5.10, 1.6.1 │ containerd: insecure handling of image volumes │
│                                  │                │          │                                      │                       │ https://avd.aquasec.com/nvd/cve-2022-23648     │
└──────────────────────────────────┴────────────────┴──────────┴──────────────────────────────────────┴───────────────────────┴────────────────────────────────────────────────┘
```

## Output of `trivy -v`:

```
Version: 0.29.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-06-17 12:07:57.264872778 +0000 UTC
  NextUpdate: 2022-06-17 18:07:57.264872578 +0000 UTC
  DownloadedAt: 2022-06-17 12:42:51.724811 +0000 UTC
```

## Additional details (base image name, container registry info...):
Base image: `google/cloud-sdk:390.0.0-alpine`
Command to install trivy
```dockerfile
ARG TRIVY_VERSION=0.29.0
RUN wget -O- https://github.com/aquasecurity/trivy/releases/download/v${TRIVY_VERSION}/trivy_${TRIVY_VERSION}_Linux-64bit.tar.gz | \
    tar -xzf - -C /
```

**Comment by tspearconquest** [comment]
This is a new vulnerability introduced by the addition of containerd support in this commit: https://github.com/aquasecurity/trivy/commit/efbc968ca8b735504c5aa7349a7770bf83d9ba6e and released in Trivy 0.29.0

**Comment by knqyf263** [comment]
Trivy is referring to this commit. As you can see, it fixes CVE-2022-23648. It means this vulnerability doesn't affect Trivy actually.
https://github.com/containerd/containerd/commit/c1bcabb45419

The problem is tagging in containerd is a bit special, then the version seems to be older than the fixed version. We will fix it so that this vulnerability will not be detected, but you don't have to worry about this vulnerability.
